perceval/archive.py

```diff
@@ -342,3 +343,123 @@ def _count_table_rows(self, table_name):
+        archive_name = hashcode[2:] + self.STORAGE_EXT
+        archive_path = os.path.join(archive_dir, archive_name)
+
+        if not os.path.exists(archive_dir):
+            os.makedirs(archive_dir)
+
+        try:
+            archive = Archive.create(archive_path)
+        except ArchiveError as e:
+            raise ArchiveManagerError(cause=str(e))
+
+        return archive
+
+    def remove_archive(self, archive_path):
+        """Remove an archive.
+
+        This method deletes from the filesystem the archive stored
+        in `archive_path`.
+
+        :param archive_path: path to the archive
+
+        :raises ArchiveManangerError: when an error occurs removing the
+            archive
+        """
+        try:
+            Archive(archive_path)
+        except ArchiveError as e:
+            raise ArchiveManagerError(cause=str(e))
+
+        os.remove(archive_path)
+
+    def search(self, origin, backend_name, category, archived_after):
+        """Search archives.
+
+        Get the archives which store data based on the given parameters.
+        These parameters define which the origin was (`origin`), how data
+        was fetched (`backend_name`) and data type ('category').
+        Only those archives created on or after `archived_after` will be
+        returned.
+
+        The method returns a list with the file paths to those archives.
+        The list is sorted by the date of creation of each archive.
+
+        :param origin: data origin
+        :param backend_name: backed used to fetch data
+        :param category: type of the items fetched by the backend
+        :param archived_after: get archives created on or after this date
+
+        :returns: a list with archive names which match the search criteria
+        """
+        archives = self._search_archives(origin, backend_name,
+                                         category, archived_after)
+        archives = [(fp, date) for fp, date in archives]
+        archives = [fp for fp, _ in sorted(archives, key=lambda x: x[1])]
+
+        return archives
+
+    def _search_archives(self, origin, backend_name, category, archived_after):
+        """Search archives using filters."""
+
+        for archive_path in self._search_files():
+            try:
+                archive = Archive(archive_path)
+            except ArchiveError:
+                continue
+
+            match = archive.origin == origin and \
+                archive.backend_name == backend_name and \
+                archive.category == category and \
+                archive.created_on >= archived_after
+
+            if not match:
+                continue
+
+            yield archive_path, archive.created_on
+
+    def _search_files(self):
+        """Retrieve the file paths stored under the base path."""
+
+        for root, _, files in os.walk(self.dirpath):
+            for filename in files:
+                location = os.path.join(root, filename)
+                yield location
```

Comment:
It is really cool this search feature to find the archives for a given data source. It could also be interesting to search just for a origin, or for a backend_name, or category. So you can say: search the archive files for `git`. But I am not sure that I have a valid use case for this feature, so then current implementation is ok for me.

Reply:
Probably that makes sense but as you say we can keep it as is and add other features in a near future.